docs/data-sources/instance_servers.md

```diff
@@ -0,0 +1,58 @@
+---
+page_title: "Scaleway: scaleway_instance_servers"
+description: |-
+Gets information about multiple Instance Servers.
+---
+
+# scaleway_instance_servers
+
+Gets information about multiple instance servers.
+
+## Examples
+
+### Basic
+
+```hcl
+# Find servers by tag
+data "scaleway_instance_servers" "my_key" {
+  tags  = ["tag"]
+}
+
+# Find servers by name and zone
+data "scaleway_instance_servers" "my_key" {
+  name = "myserver"
+  zone = "fr-par-2"
+}
+```
+
+### With single datasource
+
+```hcl
+# Find servers by tag
+data "scaleway_instance_servers" "servers_by_tag" {
+  tags  = ["tag"]
+}
+
+data "scaleway_instance_servers" "map_of_servers" {
+  for_each = {for server in data.scaleway_instance_servers.servers_by_tag.servers: server.id => server}
+  server_id = each.value.id
+}
```

Comment:
IMO we should not encourage this pattern as it would generate N additional requests for N servers. AFAIK all information available in GetServer can be retrieved from ListServers - i.e. from the data source `scaleway_instance_servers`.